mixer/pkg/runtime/routing/builder.go

```diff
@@ -503,3 +525,114 @@ func (b *builder) getBuilderAndMapperDynamic(_ ast.AttributeDescriptorFinder,
 	}
 	return instBuilder, nil, nil
 }
+
+type ruleAttributes struct {
+	parent    ast.AttributeDescriptorFinder
+	overrides map[string]*descriptor.AttributeManifest_AttributeInfo
+}
+
+func (r *ruleAttributes) GetAttribute(name string) *descriptor.AttributeManifest_AttributeInfo {
+	if info, exists := r.overrides[name]; exists {
+		return info
+	}
+	return r.parent.GetAttribute(name)
+}
+
+// buildRuleCompiler constructs an expression compiler over an extended attribute vocabulary
+// with action name prefixed template output attributes added to the global attribute manifests.
+func (b *builder) buildRuleCompiler(parent ast.AttributeDescriptorFinder, rule *config.Rule) *compiled.ExpressionBuilder {
+	// create attribute type descriptor from action templates
+	attributeDescriptor := make(map[string]*descriptor.AttributeManifest_AttributeInfo)
+	for _, action := range rule.ActionsStatic {
+		// TODO(kuat): rationalize multi-instance check actions
+		if len(action.Instances) > 0 {
```

Comment:
nit: worth extracting to a `if isCheckWithOutputTemplate(action) {` to eliminate the double-nesting here?  (and maybe just have `if !isCheckWithOutput(action) { continue }` to further reduce indent) ?

Reply:
done

Reply:
good point. I reverted the logic with `continue` to reduce the nesting.